From 52816e6e75e47f23414afde26bdce2f0d6f21feb Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Mon, 6 Jun 2016 11:26:53 -0400
Subject: [PATCH] rgmanager: Check for NULL when attempting to access the
 member list

Check that the membership list is not NULL before accessing it while sending
state information. This can happen during very shortly after startup if
modclusterd is running and sends a clustat command before initialization is
complete.

Resolves: rhbz#1228170
Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 rgmanager/src/daemons/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rgmanager/src/daemons/main.c b/rgmanager/src/daemons/main.c
index 3679d57..127015b 100644
--- a/rgmanager/src/daemons/main.c
+++ b/rgmanager/src/daemons/main.c
@@ -77,6 +77,9 @@ send_node_states(msgctx_t *ctx)
 	generic_msg_hdr hdr;
 	cluster_member_list_t *ml = member_list();
 
+	if (!ml)
+		return;
+
 	master.m_nodeid = 0;
 	event_master_info_cached(&master);
 
-- 
2.4.11

